packages/superset-ui-legacy-preset-chart-big-number/package.json

```diff
@@ -29,11 +29,11 @@
   },
   "dependencies": {
     "@data-ui/xy-chart": "^0.0.84",
-    "d3-color": "^1.2.3",
-    "prop-types": "^15.6.2",
+    "@types/shortid": "0.0.29",
     "shortid": "^2.2.14"
   },
   "peerDependencies": {
+    "d3-color": "^1.2.3",
```

Comment:
Let's keep this one as `dependencies`.

Reply:
Any reason why would it be better? I thought it's preferable to have it in `peerDependencies` as long as it is in the `dependencies` of `superset-frontend`.

Reply:
Usually `peerDependencies` should be use when only one instance of that lib should exist in the entire app. This could be due to the use of singleton or because the dependency library is very heavy. The downside of `peerDependencies` is it put more responsibility for the app developers who are consuming the library.

* Need to add that library to the app-level `dependencies`. 
* If this lib stop using `d3-color`, app developers need to remove it from their app's `dependencies` or it will become redundant dependencies.
* If the app needs `d3-color` version that is not compatible with this library, then there is a conflict.

So if the library is small and does not have singleton requirement, we should list as `dependencies` and rely on `npm` or `yarn` to handle that. This will encapsulate the dependencies and avoid the additional task for app developers to manage `d3-color` as its `dependencies`.

Reply:
Understood. And this totally makes sense. I'm wondering whether we can issue an "official" list of `peerDependencies` supported by Superset that plugin developers can trust will always be available and stable. I'd imagine `d3` packages to be one of them.

Reply:
> "official" list of peerDependencies supported by Superset that plugin developers can trust will always be available and stable. I'd imagine d3 packages to be one of them.

Each plugin may be developed on current version of d3 but then later on if d3 upgrades over time, but some plugins did not migrate, some did, it will cause problem what version Superset should say official, because everybody tries to point to the same d3. At this point, it is not clear what value we actually gain from having d3 as `peerDependencies`. It is better to use `dependencies`. 

Also, these plugins can also be used in another app that is not Superset.

Reply:
I'd say packing multi versions of d3 to the client side is still something we should actively try to avoid. Maybe incorporate something like [InspectPack](https://github.com/formidablelabs/inspectpack/) would help.